packages/sdks/src/scripts/init-editing.ts

```diff
@@ -23,7 +23,12 @@ export const registerInsertMenu = () => {
   });
 };
 
+let isSetupForEditing = false;
 export const setupBrowserForEditing = () => {
+  if (isSetupForEditing) {
+    return;
+  }
+  isSetupForEditing = true;
```

Comment:
Why is this needed? Asking because these side-effects might not consistently work for all framework environments, so if preferable I'd rather not rely on them (or know exactly what they're needed for at least)

Reply:
we could move to a window var - the point though is just that we want to be sure this only ever runs one time, not multiple, or we could face issues

Reply:
`init-editing` runs in the `onMount` hook in every framework besides RSC though, so it's guaranteed to only run once per `RenderContent`. If this is RSC-specific, can probably add a piece of boolean state in `BuilderEditingWrapper` to make sure it only runs once.

Reply:
that's exactly the problem, many people will have multiple <RenderContent /> components on one page, we don't want all this stuff to run multiple times (e.g. the message event binding for code execution - bad for perf and reliability to have these register multiple times)

Reply:
also tmk this would not be an issue with bundlers, the issue with those is tree shaking, but the "side effect" is just a local var, I've never seen a case where this is problem with even the most aggressive tree shakers

Reply:
Fair enough, we can go with this approach. 👍🏽